885 sprite blocks

dethe opened the pull request:
This isn't working completely yet, but I want to pull it anyway because it re-organizes the runtime.js file so we no longer need to separate `runtime` from `runtime-simple` and alphabetizes the sections in runtime. Aside from sprites not completely working yet, if it breaks anything let me know. I'm trying to build a concept of "drawable" that includes shapes, images, and text, so that sprites can be built from any of them.

eddieantonio commented:
So it should be noted that this code introduces a refactor, in that the redundant `runtime-simple.js` has been once again consolidated into `runtime.js` -- consequently, we're testing from `runtime.js` in QUnit.

Right after all blocks are merged, all of the remaining blocks defined in `runtime-simple.js` should be moved to `runtime.js`, and the former file should be deleted.

eddieantonio commented on js/runtime.js:
```diff
@@ -1,17 +1,36 @@
+            if (!_canvas){
+                _canvas = document.createElement('canvas');
+                _canvas.setAttribute('width', '200');
+                _canvas.setAttribute('height', '200');
```
Maybe there should be a warning that the canvas element that is used throughout is not attached to the DOM when it is created here. It might be confusing otherwise.

dethe commented:
Yes, I had removed runtime-simple, but forgot to push the commit. I also added a note about the disconnected canvas -- it is only there to help with running in the test environment.

eddieantonio commented on js/util.js:
```diff
@@ -478,6 +482,67 @@
+    }
+
+    function defaultDrawable(ctx){
+        // ctx.save();
```
What is the purpose of these commented-out context state saving/restoring methods here?

dethe replied:
Ah, I had that in there and was testing if it was needed. Doesn't appear to be needed, can probably just delete those lines.

eddieantonio commented:
> Yes, I had removed runtime-simple, but forgot to push the commit. 

Ouigh! I fear for the upcoming merge conflicts! Perhaps keep it for now; once all blocks are merged it should be safe to delete the file without merge conflicts. Unless you already have a plan for that?

azlen commented:
In theory Git will notice that the code moved to a different file. But if
not I can help with the merge. Would rather we merge back into the right
file than have to keep track of which bits have changed in one file vs. the
other.

On Fri, Jan 30, 2015 at 9:25 AM, Eddie Antonio Santos <
notifications@github.com> wrote:

> Yes, I had removed runtime-simple, but forgot to push the commit.
>
> Ouigh! I fear for the upcoming merge conflicts! Perhaps keep it for now;
> once all blocks are merged it should be safe to delete the file without
> merge conflicts. Unless you already have a plan for that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/waterbearlang/waterbear/pull/954#issuecomment-72237220>
> .
>
